Clean up styles for older browsers

dvdherron opened the pull request:
![](https://source.unsplash.com/featured/?cute,animal&CHANGE_ME)
<!-- Swap `CHANGE_ME` for a random string to get a different image -->


<!--
## Description
Uncomment if you want to provide a custom description.
-->

## Related Issue(s)
Fixes #205 



## Steps to test/reproduce
-  On Sauce Labs, view the demo page in Safari 15 and 16 (Ventura OS) or versions of other browsers before they implemented cascade layers. 
- Compare the the deploy preview here to the live site. In the latest versions of browsers, everything should be the same (except for the updated `@layers` popover example). 

![image](https://github.com/user-attachments/assets/5664180f-9c80-40af-a7ec-a102a817e24a)


_Please explain how to best reproduce the issue and/or test the changes locally (including the pages/URLs/views/states to review)._

dvdherron commented:
@stacyk I've removed the references here to layers (and most nesting). There's something happening that's causing a lot of the layout styles to not be applied. Testing on FF 95, custom properties are valid but not all of them are reflected in the styles of the page. Can you see the same thing in Safari 15?

dvdherron commented:
@stacyk I figured out why the size variables were not working. It's due to the `cq` units in the base sizes.

stacyk commented:
@dvdherron I have Safari 18 and there isn't a way to downgrade that I know of, so using Saucelabs is the way to go for that. Sounds like you are on the right trail so let me know if you want a review when ready.

dvdherron commented:
@jgerigmeyer @stacyk I think this is ready for review. 

I removed the references to layers (except for the demo example). I changed the example from having a red background to having a red border. With the former, an old browser could not read the text (It was white text on a white background). 

I also removed nesting and added some `@support` for container query related styles. Its not perfect in older browsers, but I'm not sure how far back we want to fully support for this page. 

If you test this page on Safari 16, it seems to only work when you choose `Ventura` as an OS version. 

The shadowed popovers in older browsers don't seem to be getting the "UA Stylesheet" styles we are injecting with JS. I think that's because we are using `.innerHTML` to create those elements @jgerigmeyer? These popovers aren't positioned so they push content around when they are opened.

dvdherron commented:
> Looks good to me!
> 
> > The shadowed popovers in older browsers don't seem to be getting the "UA Stylesheet" styles we are injecting with JS. I think that's because we are using `.innerHTML` to create those elements @jgerigmeyer? These popovers aren't positioned so they push content around when they are opened.
> 
> Yeah, I think this is directly related to [this note](https://github.com/oddbird/popover-polyfill/blob/main/README.md?plain=1#L140-L143). I think I fixed it (conditionally re-injecting the styles in shadow roots) in [0f301bc](https://github.com/oddbird/popover-polyfill/commit/0f301bc618ad5c7c66b511281e489360851a5fa4) -- take a look and see what you think?

Your changes fixed what I was seeing with the shadowed popovers. Thanks! This looks good to me.

stacyk commented on css/props.css:
```diff
@@ -66,3 +69,20 @@ html {
   --ccs-accent--fg2: hsl(330deg 60% 35%);
 }
+
+@supports (font-size: 2cqi) {
```
So Safari 16 supports cqi, but it looks like it doesn't like the lh unit so spacer gets ignored. Still testing Safari 15 ... 

![Screenshot 2025-03-07 at 10 57 23 AM](https://github.com/user-attachments/assets/c3b6175b-31bb-42a8-a47f-3505a46e6b0c)
![Screenshot 2025-03-07 at 10 51 25 AM](https://github.com/user-attachments/assets/e8b174aa-d0bb-4ac3-b6dc-dd2793ab5861)

dvdherron replied:
good catch. fixed in https://github.com/oddbird/popover-polyfill/pull/236/commits/5ef1a52d3f3b1edf120ad7b6b27d287c711cdd78

Safari 16: 
https://github.com/oddbird/popover-polyfill/pull/236/commits/5ef1a52d3f3b1edf120ad7b6b27d287c711cdd78

![image](https://github.com/user-attachments/assets/de523492-2a2f-4a72-bbc5-acf8d3ad5891)

![image](https://github.com/user-attachments/assets/0c905330-9194-4626-91ed-083b0f9f8e5c)

And 15:
![image](https://github.com/user-attachments/assets/9919f3e4-2433-4d87-aba5-1d80042f4de5)

![image](https://github.com/user-attachments/assets/35ee6d06-7c4b-4505-893d-d4366a489a2b)

stacyk reviewed (commented):
Super close, just a small Safari 16 issue with I think lh units (the supports query only checks for cqi)

dvdherron commented on css/demo.css:
```diff
@@ -315,3 +315,29 @@ a[target='_blank']::after {
+    color: var(--popover-color, var(--text));
+    display: flex;
+    font-family: var(--sans-serif);
+    flex-wrap: wrap;
```
Sorry @stacyk , missed this somehow! Fixed in https://github.com/oddbird/popover-polyfill/pull/236/commits/b7626a8ae488e1451256320f9f4fe616ad4bac5d

stacyk commented on css/demo.css:
```diff
@@ -15,27 +16,27 @@ body {
-    transform: scale(1.05);
-  }
+#banner-logo [data-logo] {
+  block-size: auto;
+  height: auto;
+  inline-size: 100%;
+  width: 100%;
```
@dvdherron I didn't comment on these as I think we do actually want the more "modern" property last as it will be ignored if a browser doesn't understand it. So the order:

height
block-size
width
inline-size 

is probably what we want, but other than that,  I'd say this is good to merge!

dvdherron replied:
good catch. fixed in https://github.com/oddbird/popover-polyfill/pull/236/commits/5ebfc647404ed7cfb6db32bdf44d6ff5ce701c42 so I think this is good to merge!

dvdherron commented:
@stacyk Ok, I think this still needs one more thumbs up from you and then this is ready to merge.
